src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -369,16 +370,16 @@ default boolean canSee(final Entity entity) {
      *
      * @return The "base vehicle" of the entity vehicle riding chain
      */
-    default Optional<Value.Mutable<Entity>> baseVehicle() {
-        return this.getValue(Keys.BASE_VEHICLE).map(Value::asMutable);
+    default Value<Entity> baseVehicle() {
```

Comment:
From the `net.minecraft.world.entity.Entity`:
```java
public Entity getRootVehicle() {
    Entity var0;
    for(var0 = this; var0.isPassenger(); var0 = var0.getVehicle()) {
    }

    return var0;
}
```
At a minimum, an entity itself is considered the root vehicle.

Reply:
Again, another question that is thrown up because of how you've mirrored it. In minecraft, the base vehicle certainly is treated as one's self if you're not riding anything - a good question here, however, is "does it make sense in the context of the API"?

More to the point, from an API point of view, if an entity is not riding anything, do we want this to return the entity itself, or should it be an empty optional?

Reply:
This should remain an `Optional`, the fact Vanilla treats yourself as a vehicle when you are not riding anything is an implementation detail.